### PR TITLE
rewrite compile in bash

### PR DESCRIPTION
**Pull request by yagonobre**

### CODE REVIEW

File: bin/compile [hunk 1]
```diff
@@ -1,91 +1,76 @@
...
-ORACLE_INSTANT_CLIENT_FILENAME=ORACLE_INSTANT_CLIENT_URL.split(/\//).last
+ORACLE_INSTANT_CLIENT_FILENAME=$(echo $ORACLE_INSTANT_CLIENT_URL | cut -d/ -f5)
 
-indent "Making directory #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}"
-`mkdir -p #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR}`
+source "$BIN_DIR/utils"
 
-indent "Downloading and extracting #{ORACLE_INSTANT_CLIENT_FILENAME}"
-result = `curl #{ORACLE_INSTANT_CLIENT_URL} -L -s -o - | tar -xvz -C #{BUILD_DIR}/#{ORACLE_INSTANT_CLIENT_DIR} -f - `
+puts-step "Found an .oracle.yml"
```
Comment (drgarcia1986) [inline bin/compile:21]: What about an if here to check if that file exists?

File: bin/compile [hunk 2]
```diff
@@ -1,91 +1,76 @@
...
-  end
+  SQLNET_FILENAME=$(cat $ORACLE_CONFIG | grep sqlnet.ora | cut -d' ' -f2)
+  if [[ -f $BUILD_DIR/$SQLNET_FILENAME ]]; then
+    indent "Symlinking ${SQLNET_FILENAME} to ${ORACLE_INSTANT_CLIENT_DIR}/network/admin/sqlnet.ora"
+    ln -sf $BUILD_DIR/$SQLNET_FILENAME ${ORACLE_NETWORK_ADMIN_DIR}/sqlnet.ora
+  fi
 else
-  warning '.oracle.yml is empty or invalid'
-end
+  puts-warn ".oracle.yml is empty or invalid"
```
Comment (drgarcia1986) [inline bin/compile]: What do you think about a more friendly message, like "There is no sqlnet.ora entry on .oracle.yml" ? (just to clarify WTF?!? is wrong with .oracle.yml)

**Review by drgarcia1986 — APPROVED** [review]

File: bin/compile (hunk 1 above)
Comment (drgarcia1986) [reply bin/compile:21]: (I know, there is the `detect` step, but, I don't know if with multi-buildpacks deploy the detect phase will run)

Comment (yagonobre) [reply bin/compile:21]: if the `oracle.yml` don't exist the buildpack fail, even with multi-buildpacks

File: bin/compile (hunk 2 above)
Comment (yagonobre) [reply bin/compile]: Make sense, maybe `.oracle.yml is empty`? This is the unique case tested

Comment (yagonobre) [reply bin/compile]: BTW I'm not sure if it this warning is needed, the buildpack will work fine with an empty file.

File: bin/compile (hunk 1 above)
Comment (drgarcia1986) [reply bin/compile:21]: LGTM 👍

File: bin/compile (hunk 2 above)
Comment (drgarcia1986) [reply bin/compile]: 🤔 IMO it's ok to remove warn

File: bin/compile [hunk 3]
```diff
@@ -1,91 +1,76 @@
...
+#!/usr/bin/env bash
 
-require 'yaml'
+# Usage:
+#
+#     $ bin/compile <build-dir> <env-path>
 
-$stdout.sync = true
+# Fail fast and fail hard.
+set -eo pipefail
```
Comment (aguerra) [inline bin/compile:8]: ❤️

File: bin/compile [hunk 4]
```diff
@@ -1,91 +1,76 @@
...
-  indent "Failure while downloading Oracle instant client archive: #{$?}"
+indent "Making directory ${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR}"
+mkdir -p ${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR}
+
+indent "Downloading and extracting ${ORACLE_INSTANT_CLIENT_FILENAME}"
+curl ${ORACLE_INSTANT_CLIENT_URL} -L -s -o - | tar -xz -C \
+	${BUILD_DIR}/${ORACLE_INSTANT_CLIENT_DIR} -f -
+
+if [[ $? != 0 ]]; then
+  indent "Failure while downloading Oracle instant client archive: $?"
```
Comment (aguerra) [inline bin/compile]: $? is just a code, don't know if it helps the user...

File: bin/compile [hunk 5]
```diff
@@ -1,91 +1,76 @@
...
+mkdir -p ${ORACLE_NETWORK_ADMIN_DIR}
 
-if ORACLE_CONFIG
-  TNSNAMES_FILENAME = ORACLE_CONFIG['tnsnames.ora']
-  if File.exists?(File.join(BUILD_DIR,TNSNAMES_FILENAME))
-    indent "Symlinking #{TNSNAMES_FILENAME} to #{ORACLE_INSTANT_CLIENT_DIR}/network/admin/tnsnames.ora"
-    `ln -sf #{File.join('/app',TNSNAMES_FILENAME)} #{ORACLE_NETWORK_ADMIN_DIR}/tnsnames.ora`
-  end
+if [[ -s $ORACLE_CONFIG ]]; then
+  TNSNAMES_FILENAME=$(cat $ORACLE_CONFIG | grep tnsnames.ora | cut -d' ' -f2)
```
Comment (aguerra) [inline bin/compile]: You can avoid  the cat by using `grep tnsnames.ora $ORACLE_CONFIG`

**Review by aguerra — APPROVED** [review]

File: bin/compile (hunk 4 above)
Comment (yagonobre) [reply bin/compile]: make senses

File: bin/compile (hunk 5 above)
Comment (yagonobre) [reply bin/compile]: nice!

File: bin/compile (hunk 2 above)
Comment (yagonobre) [reply bin/compile]: removed

**Comment by yagonobre** [comment]
@drgarcia1986 @aguerra updated

**Review by drgarcia1986 — APPROVED** [review]
🔁

**Review by aguerra — APPROVED** [review]
🔁

**Pull request merged**